frame/staking/src/pallet/impls.rs

```diff
@@ -783,17 +784,47 @@ impl<T: Config> Pallet<T> {
 			};
 
 			if let Some(Nominations { targets, .. }) = <Nominators<T>>::get(&voter) {
+				// if this voter is a nominator:
 				let voter_weight = weight_of(&voter);
 				if !targets.is_empty() {
+					// select only targets allowed by voter's nomination quota.
+					let nominations_quota = T::NominationsQuota::get_quota(voter_weight.into());
+
+					// if the current number of targets exceeds the nomination quota, emit an event
+					// but proceed without truncating or failing. The nomination quota checks are
+					// lazy and only fail when setting/updating the nominations.
+					if targets.len() > nominations_quota as usize {
+						Self::deposit_event(Event::<T>::NominationsQuotaExceeded {
+							staker: voter.clone(),
+							exceeded_by: (targets.len() as u32 - nominations_quota).into(),
+						});
+					}
+
+					if voters_size_tracker.try_register_voter(targets.len(), voter_bounds).is_err()
+					{
+						// no more space left for the election result, stop iterating.
+						Self::deposit_event(Event::<T>::SnapshotVotersSizeExceeded {
```

Comment:
Provided the nominator does not  amend their exceeded-nominator list, is this event continuously fired every time a snapshot is created until they correct it? If so these events could pile up if we have lazy nominators & no UI (or permissionless call) to correct it.

Reply:
This comment was actually for a few lines up -  relating to the `NominationsQuotaExceeded` event.